system-helper/flatpak-system-helper.c

```diff
@@ -1151,6 +1289,215 @@ handle_run_triggers (FlatpakSystemHelper   *object,
   return TRUE;
 }
 
+static gboolean
+get_connection_uid (GDBusMethodInvocation *invocation, uid_t *uid_out, GError **error)
+{
+  GDBusConnection *connection = g_dbus_method_invocation_get_connection (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  g_autoptr(GVariant) dict = NULL;
+  g_autoptr(GVariant) unixUserId = NULL;
+  g_autoptr(GVariant) credentials = NULL;
+  uid_t uid;
+
+   credentials = g_dbus_connection_call_sync (connection,
+                                             "org.freedesktop.DBus",
+                                             "/org/freedesktop/DBus",
+                                             "org.freedesktop.DBus",
+                                             "GetConnectionCredentials",
+                                             g_variant_new ("(s)", sender),
+                                             G_VARIANT_TYPE ("(a{sv})"), G_DBUS_CALL_FLAGS_NONE, G_MAXINT, NULL, error);
+  if (credentials == NULL)
+    return FALSE;
+
+   dict = g_variant_get_child_value (credentials, 0);
+  unixUserId = g_variant_lookup_value (dict, "UnixUserID", G_VARIANT_TYPE_UINT32);
+
+   if (unixUserId == NULL)
+    {
+      g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                   "Failed to query UnixUserId for the bus name: %s", sender);
+      return FALSE;
+    }
+
+   *uid_out = g_variant_get_uint32 (unixUserId);
+  return TRUE;
+}
+
+
+static void
+revokefs_child_setup (gpointer user_data)
+{
+  struct passwd *passwd = getpwnam("flatpak");
```

Comment:
nitpick: missing space before (

Reply:
This is worth a comment in the release notes: packaging needs to ensure that user exists

Also: there should be some form or error handlng here if it doesn't

Reply:
I expect distributors will want this username to be a configure-time parameter (and perhaps a way to turn this functionality off?)

Reply:
Not marking this as resolved as to keep a reminder here about the `flatpak` user requirement.